src/main/java/core/basesyntax/storage/impl/StorageImpl.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax.storage.impl;
+
+import core.basesyntax.storage.Storage;
+import java.util.Collection;
+import java.util.HashMap;
+
+public class StorageImpl implements Storage {
+    private final HashMap<String, StorageEntry> entries;
+
+    public StorageImpl(HashMap<String, StorageEntry> entries) {
```

Comment:
Avoid using concrete classes in method parameters when possible. It's better to use the Map interface here as well.